Adiciona script para automação de configuração e testes

gabrielzschmitz opened the pull request:
Este Pull Request introduz um novo script `test-all.sh` que facilita a configuração do ambiente e a execução de testes automatizados.  

#### **Principais funcionalidades:**  
- Prompts interativos para instalação opcional de dependências:  
  - **Liboqs**  
  - **Liboqs-go**  
  - Repositório **go-std**  
- Configuração automática de variáveis de ambiente:  
  - `LD_LIBRARY_PATH`  
  - `PKG_CONFIG_PATH`  
- Suporte à instalação de pacotes tanto em **Ubuntu** quanto em **Arch Linux**.  
- Execução de testes no diretório `src/crypto/hybrid`, com opção para definir a duração (padrão: 3 segundos).  

#### **Motivação**  
Este script automatiza tarefas repetitivas, simplifica a configuração do ambiente e oferece maior conveniência para desenvolvedores, especialmente ao trabalhar com dependências e testes.

### **Como testar**  
1. Clone o repositório e execute o script:  
   ```bash
   ./test-all.sh
   ```  
2. Siga as instruções exibidas no terminal para instalar dependências e configurar o ambiente.  
3. Verifique se os testes são executados corretamente.  

### **Tarefas realizadas:**  
- [x] Criação do script `test-all.sh`.  
- [x] Suporte para instalação em sistemas Ubuntu e Arch Linux.  
- [x] Adição de prompts interativos para instalação opcional.  
- [x] Configuração de variáveis de ambiente.  
- [x] Execução de testes Go com controle de duração.  

### **Pontos a melhorar no script**  
1. **Melhorar a função de instalação de dependências:**  
   - Atualmente, o script usa comandos estáticos para instalar dependências, o que pode dificultar a adição de novos sistemas operacionais no futuro.  
   - Sugestão: Criar uma função genérica que use tabelas de mapeamento para associar pacotes aos sistemas suportados.  

2. **Compatibilidade com POSIX:**  
   - O script não segue estritamente as normas POSIX, o que pode limitar sua portabilidade para sistemas diferentes de Linux.  

3. **Validação e tratamento de erros mais robustos:**  
   - Certificar-se de que todos os comandos críticos sejam validados com verificações de sucesso ou falha, retornando mensagens de erro claras.  

4. **Adição de logs:**  
   - Implementar logs que registrem ações realizadas pelo script, facilitando a depuração em caso de falhas.  

5. **Melhor documentação:**  
   - Criar um README detalhado para o script, com exemplos de uso e explicações sobre o propósito de cada seção.  

6. **Possibilidade de executar em modo não interativo:**  
   - Incluir uma flag (ex.: `--non-interactive`) para permitir execução automatizada sem prompts interativos, com base em configurações pré-definidas.  

7. **Testes automatizados para o script:**  
   - Criar testes para validar as funcionalidades do script, garantindo que ele funcione corretamente em diferentes cenários e ambientes.
   
Closes #9

gabrielzschmitz committed:
feat: add test-all.sh script for automated testing and setup

This commit introduces a `test-all.sh` script to streamline the setup,
installation, and testing process. Key features include:
- Installation prompts for Liboqs, Liboqs-go, and go-std repository.
- Environment variable configuration for `LD_LIBRARY_PATH` and
  `PKG_CONFIG_PATH`.
- Integration with the go-std repository for additional setup and testing.
- Option to specify test duration for Go tests, defaulting to 3 seconds.
- Support for both Ubuntu and Arch Linux package installations.

The script improves usability by automating repetitive tasks and
providing clear prompts for optional installations.

PQC-Group-UTFPR commented:
Parece bom! Acho que só falta a linha que chamaria o teste do não-híbrido. Ou entendi errado?

gabrielzschmitz commented:
> Parece bom! Acho que só falta a linha que chamaria o teste do não-híbrido. Ou entendi errado?

Isso, quando o outro PR estiver finalizado é só adicionar aqui o `go test <path>`

alexandreboutrik committed:
feat: add support for Gentoo Linux

This commit enables the script to automatically detect the distribution
(by reading the /etc/lsb-release file), removing the need for user
interaction, and adds support for Gentoo Linux.

PQC-Group-UTFPR reviewed (approved):
Funciona (tive que fazer um go get e mudar PATH; vou testar em uma instância multipass para ter certeza). Mas aprovo. Obrigado
